twilio/request_validator.py

```diff
@@ -21,19 +21,47 @@ def compare(string1, string2):
     result = True
     for c1, c2 in izip(string1, string2):
         result &= c1 == c2
+        
     return result
 
 
 def remove_port(uri):
     """Remove the port number from a URI
 
-    :param uri: full URI that Twilio requested on your server
+    :param uri: parsed URI that Twilio requested on your server
 
     :returns: full URI without a port number
     :rtype: str
     """
+    if not uri.port:
+        return uri.geturl()
+
     new_netloc = uri.netloc.split(':')[0]
     new_uri = uri._replace(netloc=new_netloc)
+
+    return new_uri.geturl()
+
+
+def add_port(uri):
+    """Add the port number to a URI
+
+    :param uri: parsed URI that Twilio requested on your server
+    :param port: the port number to be added to the URI
+
+    :returns: full URI with a port number
+    :rtype: str
+    """
+    if uri.port:
+        return uri.geturl()
+    
+    port = None
+    if uri.scheme == "https":
+        port = 443
+    elif uri.scheme == "http":
+        port = 80
```

Comment:
What if it's neither?

Reply:
I was wondering about this... what kind of behavior would we want in that situation? Are non http/https uri's ever valid?

Reply:
I think this is fine for now: `port = 443 if uri.scheme == "https" else 80`.

If we have usecases later for default ports for other schemes they can be added.